### DISCUSSION TITLE
Covariances set to 0 in pool.compare

### DISCUSSION THREAD

**Discussion by CaroHaensch** [amices/mice#374] [Impute, analyse and pool]
Dear Stef, dear Gerko,

Jörg Drechsler and I noticed quite different results in pool.compare() between mice >3 and mice <3 for the Wald test. I think we pinned it down to the variance-covariance matrix which is not stored completely anymore. 

In line 32 of pool.compare(), the variance vector is transformed again into a matrix with diag() 		

Ubar <- Q %*% diag(est1$pooled$ubar) %*% (t(Q))

and the covariances are of course then all set to zero. 

In the old mice version it was

 Ubar <- Q %*% est1$ubar %*% (t(Q))

The difference in covariance values will lead to different results and we think in case of the newer version to incorrect ones. Li et al 1991 use the variance-covariance matrix too.

Cheers
Carolina

**Comment by stefvanbuuren** [comment]
Carolina, thanks for alerting. You are right: `pool.compare()` gives the wrong results.

`pool.compare()` should have been deprecated in `mice 3.0` in favour of `D1()` and `D3()`. I noted the problem in https://github.com/stefvanbuuren/mice/blob/master/tests/testthat/test-D1.R, but evidently forgot to deprecate. I will do so in the next release, and remove the function in due time.

My suggestion is to use `D1()` instead, which does have the full variance-covariance matrix.

Related: 
#121 

https://stackoverflow.com/questions/51224252/mice-package-in-r-mipo-object-does-not-return-variance-covariance-matrix-anymor

**Comment by stefvanbuuren** [comment]
`mice 3.8.0` deprecates the `pool.compare()` function.
